src/libsyntax/parse/parser.rs

```diff
@@ -6167,7 +6150,8 @@ impl<'a> Parser<'a> {
             let is_auto = if self.eat_keyword(keywords::Trait) {
                 IsAuto::No
             } else {
-                self.eat_auto_trait();
+                self.eat_keyword(keywords::Auto);
```

Comment:
Rule of thumb: `eat_keyword` shouldn't be used unless you use the returned result.
`expect_keyword` or simply `bump` would be better here.